src/Microsoft.MobileBlazorBindings.WebView/BlazorHybridHost.cs

```diff
@@ -36,22 +45,78 @@ public static void AddResourceAssembly(Assembly resourceAssembly, string content
+            else
+            {
+                foreach (var lazyProvider in _resourceFileProviders)
+                {
+                    var provider = lazyProvider.Value;
+
+                    try
+                    {
+                        var fileInfo = provider.GetDirectoryContents("");
+
+                        if (fileInfo != null && fileInfo.Any() && fileInfo.First().PhysicalPath.Contains(contentRoot))
+                        {
+                            // there is already a provider registered that supports this content root (mainly the android provider).
+                            return;
+                        }
+                    } finally
+                    {
+                        if (provider is IDisposable disposable)
+                        {
+                            disposable.Dispose();
+                        }
+                    }
+                }
+
+                throw new DirectoryNotFoundException($"The directory {contentRoot} is not found.");
+            }
+        }
+
+        /// <summary>
+        /// Tries to load the given <paramref name="filename"/> from a registered file provider.
         /// </summary>
-        /// <param name="filename">The filename of the resource to load. This is appended to the specified content root in a resource assembly to get the full path.</param>
-        /// <param name="fileStream">A readable stream with the resource content. Only valid if the method call returns <c>true</c>.</param>
+        /// <param name="filename">The filename of the file to load. This file is tried to retrieve from all registered file providers.</param>
+        /// <param name="fileStream">A readable stream with the FILE content. Only valid if the method call returns <c>true</c>.</param>
         /// <returns>Returns <c>true</c> if the resource file was found; <c>false</c> otherwise.</returns>
-        public static bool TryGetEmbeddedResourceFile(string filename, out Stream fileStream)
+        public static bool TryGetFile(string filename, out Stream fileStream)
         {
             foreach (var resourceFileProviderCreator in _resourceFileProviders)
             {
                 var resourceFileProvider = resourceFileProviderCreator.Value;
-                var fileInfo = resourceFileProvider.GetFileInfo(filename);
-                if (fileInfo.Exists)
+                try
+                {
+                    var fileInfo = resourceFileProvider.GetFileInfo(filename);
+                    if (fileInfo != null && fileInfo.Exists)
+                    {
+                        fileStream = fileInfo.CreateReadStream();
+                        return true;
+                    }
+                } finally
                 {
-                    fileStream = fileInfo.CreateReadStream();
-                    return true;
+                    if (resourceFileProvider is IDisposable disposable)
+                    {
+                        disposable.Dispose();
```

Comment:
Why dispose a file provider just because it failed? Maybe it failed on one file, but it would work with another?

Reply:
Must have been late. In my mind a "Lazy<>" turned into a factory.